get_mvn.go

```diff
@@ -0,0 +1,161 @@
+}
+
+func (g *MvnGetter) Get(dst string, u *url.URL) error {
+	q := u.Query()
+	artifactId := q.Get("artifactId")
+	if artifactId == "" {
+		return fmt.Errorf("query parameter 'artifactId' is required.")
+	}
+	version := q.Get("version")
+	if version == "" {
+		return fmt.Errorf("query parameter 'version' is required.")
+	}
+	classifier := q.Get("classifier")
+
+	artType := q.Get("type")
+	if artType == "" {
+		artType = "jar"
+	}
+
+	filename := artifactId + "-" + version
+	if classifier != "" {
+		filename += "-" + classifier
+	}
+	filename += "." + artType
+
+	dstFile := filepath.Join(dst, filename)
+	return g.GetFile(dstFile, u)
+}
+
+// Get the remote file.
+// The created filename will always be '<artifactId>-<version>[-<classifier>].<type>', Ex., 'testng-6.13.1.jar'. So the name in the `client` passed arg 'dst' is ignored.
+//
+// If the version is a snapshot version, it will get the latest snapshot artifact.
+// Query parameters:
+//   - groupId: the group id
+//   - artifactId: the artifact id
+//   - version: the artifact version
+//   - type: the artifact type, default as 'jar'
+// example url: mvn::http://username@host/mavan/repo/path?groupId=org.example&artifactId=test&version=1.0.0-SNAPSHOT
+func (g *MvnGetter) GetFile(dst string, u *url.URL) error {
+	q := u.Query()
+	groupId := q.Get("groupId")
+	if groupId == "" {
+		return fmt.Errorf("query parameter 'groupId' is required.")
+	}
+	artifactId := q.Get("artifactId")
+	if artifactId == "" {
+		return fmt.Errorf("query parameter 'artifactId' is required.")
+	}
+	version := q.Get("version")
+	if version == "" {
+		return fmt.Errorf("query parameter 'version' is required.")
+	}
+	classifier := q.Get("classifier")
+	artType := q.Get("type")
+	if artType == "" {
+		artType = "jar"
+	}
+
+	artifactUrl, err := url.Parse(u.String())
+	if err != nil {
+		return err
+	}
+	artifactUrl.RawQuery = ""
+	artifactUrl.Path += fmt.Sprintf("/%s/%s/%s", strings.Replace(groupId, ".", "/", -1), artifactId, version)
+
+	ver := version
+	if strings.HasSuffix(version, "-SNAPSHOT") {
+		// get the latest snapshot
+		snapshotVer, err := g.ParseLastestSnapshotVersion(artifactUrl)
+		if err != nil {
+			return err
+		}
+
+		ver = snapshotVer
+	}
+
+	filename := artifactId + "-" + ver
+	if classifier != "" {
+		filename += "-" + classifier
+	}
+	filename += "." + artType
+	artifactUrl.Path += "/" + filename
+	dstFile := dst
+	// if it's not auto decompress archive mode, use the real file name
+	if !strings.HasSuffix(dst, "/archive") {
+		dstFile = filepath.Join(filepath.Dir(dst), filename)
+	}
+
+	log.Printf("Downloading %s to %s", artifactUrl, dstFile)
+	return g.HttpGet.GetFile(dstFile, artifactUrl)
+}
+
+func (g *MvnGetter) ParseLastestSnapshotVersion(artifactUrl *url.URL) (string, error) {
+	mvnMetaUrl, err := url.Parse(artifactUrl.String())
+	if err != nil {
+		return "", err
+	}
+	mvnMetaUrl.Path += "/maven-metadata.xml"
+
+	mvnMetaFile, err := ioutil.TempFile(os.TempDir(), "maven-metadata")
```

Comment:
You can just use an empty string instead of TempDir and TempFile will do the right thing: `ioutil.TempFile("", ...)`